File: javascript/7-Reverse-Integer.js
/**
 * @param {number} x
 * @return {number}
 */
const reverse = function (x) {
  const max = 2 ** 31 - 1;
  const min = -(2 ** 31);

  let result = 0;
  while (x !== 0) {
    const digit = x % 10;
    x = Math.trunc(x / 10);

    if (result > max / 10 || (result === max / 10 && digit >= max % 10)) {
      return 0;
    } else if (
      result < min / 10 ||
      (result === max / 10 && digit <= min % 10)
    ) {
      return 0;
    } else {
      result = result * 10 + digit;
    }
  }

  return result;
};

File: swift/7-Reverse-Integer.swift
class Solution {
    func reverse(_ x: Int) -> Int {
        let MIN = Int32.min
        let MAX = Int32.max
        
        var res = 0
        var value = x
        while value != 0 {
            let digit = value % 10
            value /= 10
            
            if res > MAX / 10 || (res == MAX / 10 && digit >= MAX % 10) {
                return 0
            }
            if res < MIN / 10 || (res == MIN / 10 && digit <= MIN % 10) {
                return 0
            }
            res = (res * 10) + digit 
        }
        return res
    }
}